watertap/unit_models/tests/test_clarifier.py

```diff
@@ -283,3 +299,445 @@ def configure(self):
+        m.fs.unit,
+        submodel_scalers={
+            m.fs.unit.mixed_state: ASM1PropertiesScaler,
+            m.fs.unit.underflow_state: ASM1PropertiesScaler,
+            m.fs.unit.effluent_state: ASM1PropertiesScaler,
+        },
+    )
+
+
+def scale_vars_with_iscale(m):
+    # Set scaling factors for badly scaled variables
+    iscale.set_scaling_factor(m.fs.unit.underflow_state[0.0].pressure, 1e-5)
+    iscale.set_scaling_factor(
+        m.fs.unit.underflow_state[0.0].conc_mass_comp["X_BA"], 1e3
+    )
+    iscale.set_scaling_factor(m.fs.unit.underflow_state[0.0].conc_mass_comp["X_P"], 1e3)
+    iscale.set_scaling_factor(m.fs.unit.underflow_state[0.0].conc_mass_comp["S_O"], 1e3)
+    iscale.set_scaling_factor(
+        m.fs.unit.underflow_state[0.0].conc_mass_comp["S_NO"], 1e3
+    )
+    iscale.set_scaling_factor(m.fs.unit.effluent_state[0.0].pressure, 1e-5)
+    iscale.set_scaling_factor(m.fs.unit.effluent_state[0.0].conc_mass_comp["X_BA"], 1e7)
+    iscale.set_scaling_factor(m.fs.unit.effluent_state[0.0].conc_mass_comp["X_P"], 1e7)
+    iscale.set_scaling_factor(m.fs.unit.effluent_state[0.0].conc_mass_comp["S_O"], 1e7)
+    iscale.set_scaling_factor(m.fs.unit.effluent_state[0.0].conc_mass_comp["S_NO"], 1e7)
+
+    iscale.calculate_scaling_factors(m.fs.unit)
+
+
+def perturb_solution(m):
+    m.fs.unit.inlet.flow_vol.fix(18446 * 0.8 * units.m**3 / units.day)
+    m.fs.unit.split_fraction[0, "effluent", "H2O"].fix(0.993 * 0.35)
+    m.fs.unit.inlet.conc_mass_comp[0, "S_I"].fix(27 * 1.5 * units.g / units.m**3)
+
+
+@pytest.mark.requires_idaes_solver
+@pytest.mark.unit
+def test_scaling_profiler_with_scalers():
+    sp = ScalingProfiler(
+        build_model=build_model,
+        user_scaling=scale_vars_with_scalers,
+        perturb_state=perturb_solution,
+    )
+
+    stream = StringIO()
+
+    sp.report_scaling_profiles(stream=stream)
+
+    expected = """
+============================================================================
+Scaling Profile Report
+----------------------------------------------------------------------------
+Scaling Method           || User Scaling           || Perfect Scaling
+Unscaled                 || 6.241E+06 | Solved 3   ||
+Vars Only                || 1.359E+10 | Solved 3   || 1.356E+14 | Solved 1  
+Harmonic                 || 1.359E+10 | Solved 3   || 1.228E+02 | Solved 1  
+Inverse Sum              || 1.359E+10 | Solved 3   || 6.636E+03 | Solved 1  
+Inverse Root Sum Squares || 1.359E+10 | Solved 3   || 6.633E+03 | Solved 1  
+Inverse Maximum          || 1.359E+10 | Solved 3   || 6.636E+03 | Solved 1  
+Inverse Minimum          || 1.359E+10 | Solved 3   || 9.327E+01 | Solved 1  
+Nominal L1 Norm          || 1.359E+10 | Solved 3   || 1.817E+03 | Solved 1  
+Nominal L2 Norm          || 1.359E+10 | Solved 3   || 3.354E+03 | Solved 1  
+Actual L1 Norm           || 1.359E+10 | Solved 3   || 9.450E+01 | Solved 1  
+Actual L2 Norm           || 1.359E+10 | Solved 3   || 8.480E+01 | Solved 1  
```

Comment:
remind me again--are the values here condition numbers?

Reply:
Yes and yes